### PR TITLE
Dev2

### PR DESCRIPTION
**Pull request by dev5151**
1) Switched from Custom Login to Firebase Auth 
2) Added Card View For Login and Sign up.

### CODE REVIEW

File: app/src/main/java/com/exuberant/code2create/activities/HomeActivity.java [hunk 1]
```diff
@@ -284,6 +285,8 @@ void setupMenuListeners() {
             SharedPreferences.Editor editor = sharedPreferences.edit();
             editor.clear();
             editor.apply();
+            FirebaseAuth mAuth=FirebaseAuth.getInstance();
```
Comment (reallyinvincible) [inline app/src/main/java/com/exuberant/code2create/activities/HomeActivity.java]: Give space before and after =

File: app/src/main/java/com/exuberant/code2create/activities/LoginActivity.java [hunk 2]
```diff
@@ -35,83 +52,109 @@
...
 
     Button loginButton;
     EditText emailET, passwordET;
-
+    CardView cardView;
+    ProgressBar progressBar;
+    String uid;
+    TextView tvLogin, tvSentence;
+    LinearLayout linerarLayout1;
+    Button signInButton;
```
Comment (reallyinvincible) [inline app/src/main/java/com/exuberant/code2create/activities/LoginActivity.java]: Fix typos and follow naming conventions

File: app/src/main/java/com/exuberant/code2create/activities/HomeActivity.java (hunk 1 above)
Comment (reallyinvincible) [inline app/src/main/java/com/exuberant/code2create/activities/HomeActivity.java]: Give space before and after =

File: app/src/main/java/com/exuberant/code2create/activities/LoginActivity.java [hunk 3]
```diff
@@ -35,83 +52,109 @@
...
     List<Agenda> agendaList;
     AgendaModel model;
 
     ConstraintLayout constraintLayout;
 
+    public LoginActivity() {
+
+    }
+
```
Comment (reallyinvincible) [inline app/src/main/java/com/exuberant/code2create/activities/LoginActivity.java]: Why there is an empty constructor for an activity?

File: app/src/main/java/com/exuberant/code2create/activities/LoginActivity.java [hunk 4]
```diff
@@ -35,83 +52,109 @@
...
         }
+
         loginButton.setOnClickListener(view -> {
+            signIn();
+
+        });
+
+        tvLogin.setOnClickListener(v -> {
+            tvClicked();
```
Comment (reallyinvincible) [inline app/src/main/java/com/exuberant/code2create/activities/LoginActivity.java]: Name the method appropriately with what actions it does

File: app/src/main/java/com/exuberant/code2create/adapters/AgendaAdapter.java [hunk 5]
```diff
@@ -9,14 +10,24 @@
...
 
 import androidx.annotation.NonNull;
 import androidx.recyclerview.widget.RecyclerView;
 
 public class AgendaAdapter extends RecyclerView.Adapter<AgendaAdapter.AgendaAdapterViewHolder> {
 
     List<Agenda> agendaList;
+    String endTime;
+    String startTime;
+    Integer var2,var1;
```
Comment (reallyinvincible) [inline app/src/main/java/com/exuberant/code2create/adapters/AgendaAdapter.java]: what is var1 and var2? Name variables appropriately

File: app/src/main/java/com/exuberant/code2create/fragments/FoodCouponsFragment.java [hunk 6]
```diff
@@ -193,6 +197,8 @@ private void initialiseViews(View view) {
         cardView1 = view.findViewById(R.id.cv_coupon1);
         cardView2 = view.findViewById(R.id.cv_coupon2);
         cardView3 = view.findViewById(R.id.cv_coupon3);
+        mAuth=FirebaseAuth.getInstance();
+        uid=mAuth.getUid();
```
Comment (reallyinvincible) [inline app/src/main/java/com/exuberant/code2create/fragments/FoodCouponsFragment.java]: Fix spacings everywhere

File: app/src/main/java/com/exuberant/code2create/fragments/FaqsFragment.java [hunk 7]
```diff
@@ -1,70 +1,52 @@
...
+        faqsList.add(new FaqsModel(R.string.faq_q4,R.string.faq_a4));
+        faqsList.add(new FaqsModel(R.string.faq_q5,R.string.faq_a5));
+        faqsList.add(new FaqsModel(R.string.faq_q6,R.string.faq_a6));
+        faqsList.add(new FaqsModel(R.string.faq_q7,R.string.faq_a7));
+        faqsList.add(new FaqsModel(R.string.faq_q8,R.string.faq_a8));
+        faqsList.add(new FaqsModel(R.string.faq_q9,R.string.faq_a9));
+        faqsList.add(new FaqsModel(R.string.faq_q10,R.string.faq_a10));
+        faqsList.add(new FaqsModel(R.string.faq_q11,R.string.faq_a11));
+
+        FaqsAdapter faqsAdapter  =new FaqsAdapter(faqsList);
```
Comment (reallyinvincible) [inline app/src/main/java/com/exuberant/code2create/fragments/FaqsFragment.java]: Inconsistent spaces

File: app/src/main/res/layout/activity_login.xml [hunk 8]
```diff
@@ -10,18 +10,21 @@
     tools:context=".activities.LoginActivity">
 
     <ImageView
+        android:id="@+id/imageView7"
```
Comment (reallyinvincible) [inline app/src/main/res/layout/activity_login.xml]: Give meaningful names

File: app/src/main/res/layout/activity_login.xml [hunk 9]
```diff
@@ -90,10 +48,145 @@
             android:paddingTop="12dp"
             android:paddingEnd="64dp"
             android:paddingBottom="12dp"
-            android:text="Login"
+            android:text="Sign In "
```
Comment (reallyinvincible) [inline app/src/main/res/layout/activity_login.xml]: Check for extra space after text and also extract text to string resources

File: app/src/main/res/layout/activity_login.xml [hunk 10]
```diff
@@ -90,10 +48,145 @@
...
+            app:icon="@drawable/googleg_standard_color_18"
+            app:iconGravity="textStart"
+            app:iconPadding="10dp" />
 
     </LinearLayout>
 
+    <ProgressBar
+        android:id="@+id/progressBar"
+        style="?android:attr/progressBarStyle"
+        android:layout_width="89dp"
```
Comment (reallyinvincible) [inline app/src/main/res/layout/activity_login.xml]: Don't hardcode any views. Use margins and padding to achieve desired results

File: app/src/main/java/com/exuberant/code2create/fragments/FaqsFragment.java (hunk 7 above)
Comment (dev5151) [reply app/src/main/java/com/exuberant/code2create/fragments/FaqsFragment.java]: Won't happen again

File: app/src/main/java/com/exuberant/code2create/adapters/AgendaAdapter.java (hunk 5 above)
Comment (dev5151) [reply app/src/main/java/com/exuberant/code2create/adapters/AgendaAdapter.java]: Removed

File: app/src/main/java/com/exuberant/code2create/fragments/FoodCouponsFragment.java [hunk 11]
```diff
@@ -90,8 +93,9 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
         context = this.getContext();
         initialiseViews(view);
 
-        sharedPreferences = getActivity().getSharedPreferences(getString(R.string.shared_prefs_name), Context.MODE_PRIVATE);
+       /* sharedPreferences = getActivity().getSharedPreferences(getString(R.string.shared_prefs_name), Context.MODE_PRIVATE);
```
Comment (reallyinvincible) [inline app/src/main/java/com/exuberant/code2create/fragments/FoodCouponsFragment.java]: Remove code that is not being used anymore.

File: app/src/main/res/layout/fragment_prizes.xml [hunk 12]
```diff
@@ -229,6 +230,43 @@
...
+                android:text="@string/Rs2000"
+                android:textColor="@android:color/white"
+                android:textSize="20sp"
+                app:layout_constraintBottom_toBottomOf="@+id/textView10"
+                app:layout_constraintEnd_toEndOf="@+id/textView23"
+                app:layout_constraintStart_toStartOf="@+id/textView23"
+                app:layout_constraintTop_toTopOf="@+id/textView10" />
+
+            <TextView
+                android:id="@+id/textView10"
```
Comment (reallyinvincible) [inline app/src/main/res/layout/fragment_prizes.xml]: Another minor recommendation, although it's okay to use the autogenerated ids when in constraint layout, for the sake of consistency, use snake case for ids.

File: app/src/main/java/com/exuberant/code2create/activities/HomeActivity.java [hunk 13]
```diff
@@ -54,7 +54,7 @@
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-//        getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE, WindowManager.LayoutParams.FLAG_SECURE);
+
```
Comment (reallyinvincible) [inline app/src/main/java/com/exuberant/code2create/activities/HomeActivity.java]: Don't remove flag just uncomment it

File: app/src/main/java/com/exuberant/code2create/activities/LoginActivity.java [hunk 14]
```diff
@@ -36,282 +41,111 @@
...
-    LinearLayout googleAuthLinearLayout;
-    Button signInButton;
-    List<Agenda> agendaList;
-    AgendaModel model;
+    @Override
+    protected void onStart() {
+        super.onStart();
 
-    ConstraintLayout constraintLayout;
+        if (loginState==1) {
```
Comment (reallyinvincible) [inline app/src/main/java/com/exuberant/code2create/activities/LoginActivity.java]: Inconsistent spacing

File: app/src/main/java/com/exuberant/code2create/activities/LoginActivity.java [hunk 15]
```diff
@@ -36,282 +41,111 @@
...
-                        if (registeredEmail.equals(email)) {
-                            uid = mAuth.getUid();
-                            mUserReference.child(uid).setValue(email);
-                            showConfirmationSnackbar("Sign In Success");
-                        } else {
-                            showErrorSnackbar("User does not exist in Database");
-                        }
-                    }
-                }
+            public void switchToSignIp() {
```
Comment (reallyinvincible) [inline app/src/main/java/com/exuberant/code2create/activities/LoginActivity.java]: Typo in sign up

File: app/src/main/java/com/exuberant/code2create/fragments/LoginFragment.java [hunk 16]
```diff
@@ -0,0 +1,280 @@
...
+                            progressBar.setVisibility(View.GONE);
+                            showErrorSnackbar("User does not exist in Database");
+                        }
+                    }
+                }
+            }
+
+            @Override
+            public void onCancelled(@NonNull DatabaseError databaseError) {
+
```
Comment (reallyinvincible) [inline app/src/main/java/com/exuberant/code2create/fragments/LoginFragment.java]: Handle on cancelled calls everywhere by displaying error message

**Pull request merged**